commands/editmessage.js

```diff
@@ -0,0 +1,58 @@
+const Discord = require("discord.js")
+
+module.exports = async (bot, message, args) => {
+    if (args[0] == "help"){
+        let help = new Discord.RichEmbed()
+        .setColor("#FFFF00")
+        .addField("Usage", "editmessage <message_id> <message content>")
+        return message.channel.send(help).then(r => r.delete(30000))
+    }
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
+            await message.delete()
+            return m.delete(15000)
+        })
+    }
+    message_id = args[0]
+    // put all channels in the server in an array
+    let channels = message.guild.channels
+    let l = channels.array().length
+    channels = channels.first(l)
+    // channels is now an arrray of all the channels in this server
+    let i
+    // now keep looping through the text channels and try to fetch the message
+    let good_message
+    message.channel.send("Loading...").then(async m=>{
```

Comment:
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
    message.channel.send("Loading...").then(async function(m) {
```